### ISSUE TITLE
Pathfinding NMS. automated ground units lose their way

### ISSUE THREAD

**Issue by MrZorG33** [We-the-People-civ4col-mod/Mod#779]
NMS - new movement system
ground units sent to any point on the map (to a city or to any tile) in automatic mode constantly stop, as if they have lost their way. no hostile units interfere with this, since this also happens on the way when there are no hostile units.

**Comment by MrZorG33** [comment]
@devolution79 load the save. you will see one active treasure. send it to any city. after one or more moves it will stop.
I play without wild animals, so they can't be the reason for the stop. the unit "loses" the path to the destination.

[test pathfinding.zip](https://github.com/We-the-People-civ4col-mod/Mod/files/11167133/test.pathfinding.zip)

**Comment by devolution79** [comment]
Wait a sec ! What is your setting for USE_CLASSIC_MOVEMENT_SYSTEM ?

**Comment by devolution79** [comment]
I can definitely say by my recent testing that there is indeed strange behaviour related to automoves when the new movement system is enabled. In my opinion this system is in beta and is not "production ready" atm.

**Comment by raystuttgart** [comment]
@jo-oe has contributed a fix for it.

**Comment by MrZorG33** [comment]
thanks @jo-oe . I will check it in the coming days)

**Comment by MrZorG33** [comment]
While doing a test game, I discovered that Pathfinding with NMS has another problem: instead of moving to the tile I want in the direction I chose (green arrow), the unit tries to go the other way (red arrow).
it does not matter how to control the unit - keyboard or mouse.
![поиск пути  ошибка перемещения юнита](https://github.com/We-the-People-civ4col-mod/Mod/assets/61798883/c4989704-8346-49cd-b526-6c311636e9ce)

**Comment by raystuttgart** [comment]
@MrZorG33 

**Edit:**
I first thought that the City was the final destination. :)
So my first post was wrong.

------------

Still it needs to be checked in the case above if both paths e.g. have **roads**.
It is indeed possible that the red path ends with **more movement points left** or equal **movement points** as the green in case of roads.

Even in case it both paths require same amount of movement points it would not really be a bug as pathfinding prioritizes movement through Cities.
(As those are generally considered to be save Plots.)


**Summary:**
It would only be a bug if the red path consumes more Movement Points than the green.
If both consume the same amount of movement Points or the red one consumes less, than it is not.

**In other words:**
If the red path has roads and the green one does not, it is possible that there is no bug.

**Comment by MrZorG33** [comment]
@raystuttgart the road is only on the tile with the number 2 (and, of course, in the city).
If I remember correctly (I'll try to find the game save if it wasn't overwritten), the 2 tile has a move cost of 4, and the 1 tile has a move cost of 3. however, if moving from tile 1, you must first move to the city on a hill (movement cost 2) and only then move to tile 2. so this movement vector is not beneficial anyway.

I cannot say exactly when this bug appeared, but I am sure that not immediately after the introduction of NMS.

that is, if you move along the red vector, you will spend 5 turns to arrive at the tile with number 2. and if you move along the green vector, then only 4 turns.

I'll try to find a save for this game by the end of this week.
although this situation can be modeled with the help of WB.

**Comment by raystuttgart** [comment]
Well, it definitely sounds like a bug then in the Pathfinding for "New Movement System".  
Let us see then if somebody finds time to fix and analyse.

**Comment by devolution79** [comment]
The problem is pathCost which computes a negative cost for certain moves. The https://github.com/We-the-People-civ4col-mod/Mod/tree/fix_pathfinding fixes it I think (but it's been ages since I touched it and don't recall all the details :( )

**Comment by MrZorG33** [comment]
@devolution79, could you test this fix?

**Comment by devolution79** [comment]
@MrZorG33  I need someone to test it since I am overwhelmed at the moment. Please check out the branch and try to reproduce the problem.

**Comment by MrZorG33** [comment]
@devolution79 Which branch should I test?

**Comment by MrZorG33** [comment]
reopened - problem still present.
